core/src/main/java/org/infinispan/transaction/impl/AbstractEnlistmentAdapter.java

```diff
@@ -79,7 +85,7 @@ protected final void releaseLocksForCompletedTransaction(LocalTransaction localT
    }
 
    private void removeTransactionInfoRemotely(LocalTransaction localTransaction, GlobalTransaction gtx) {
-      if (mayHaveRemoteLocks(localTransaction) && !isSecondPhaseAsync) {
+      if (mayHaveRemoteLocks(localTransaction) && !isSecondPhaseAsync && partitionHandlingManager.isTransactionPartiallyRolledBack(gtx)) {
```

Comment:
This doesn't read very well, if partition handling is disabled I would expect to never have partially rolled back txs.